src/app/shared/example-viewer/example-viewer.html

```diff
@@ -52,10 +52,10 @@
         </mat-icon>
       </button>
     </div>
-    <code-snippet [source]="file" [lines]="lines" *ngIf="file !== 'demo'"></code-snippet>
+    <code-snippet [source]="generateUrl(file)" [lines]="lines"></code-snippet>
```

Comment:
Doing it this way means we have to redo the work to generate the URL on every change detection. Instead you can make the file input a getter/setter and do the URL generation in the setter.

i.e. in the TS:

```ts
@Input()
get file() { return this._file; }
set file(relativeFile) {
  this._file = generateUrl(relativeFile);
}
```

and then in the HTML just:
```html
<... [source]="file" ...>
```

Reply:
I had to put it in `ngOnInit` otherwise `this.example` and `this.exampleData.module.importSpecifier` would be missing